modules/datahandling/projectrequests.go

```diff
@@ -196,8 +237,22 @@ type projectRevokePermissionsRequest struct {
 }
 
 func (p projectRevokePermissionsRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
-	// TODO: check if permission high enough on project
-	err := db.MySQLProjectRevokePermission(p.ProjectID, p.RevokeUsername, p.SenderID)
+	hasPermission, err := dbfs.PermissionAtLeast(p.SenderID, p.ProjectID, "admin", db)
+	if err != nil {
+		utils.LogError("API permission error", err, utils.LogFields{
+			"Resource":  p.Resource,
+			"Method":    p.Method,
+			"SenderID":  p.SenderID,
+			"ProjectID": p.ProjectID,
+		})
+		return []dhClosure{toSenderClosure{msg: messages.NewEmptyResponse(messages.StatusFail, p.Tag)}}, nil
+	}
+	// allow case where user is removing themselves from a project
+	if !hasPermission && p.SenderID != p.RevokeUsername {
+		return []dhClosure{toSenderClosure{msg: messages.NewEmptyResponse(messages.StatusUnauthorized, p.Tag)}}, nil
+	}
+
```

Comment:
because owner permissions are stored separately from normal permissions, we don't have to worry about this case here, as it will fail on the db and tell us about it. If we want to throw a different message for it, I can change this to check that explicitly.